manticore/core/smtlib/solver.py

```diff
@@ -351,7 +351,18 @@ def get_all_values(self, constraints, expression, maxcnt=300, silent=False):
             while self._check() == 'sat':
                 value = self._getvalue(var)
                 result.append(value)
-                self._assert( var != value )
+
+                # Reset the solver to avoid the incremental mode 
```

Comment:
The incremental mode is usually slower (see https://stackoverflow.com/questions/23736269/what-is-the-benifits-of-incremental-solving/23739636#23739636). 
In several cases it is faster to do two formulas:
```
constraints
x != v_0
check-sat
```
```
constraints
x != v_0
x != v_1
check-sat
```
Than

``` :
constraints
x != v_0
check-sat
x != v_1
check-sat
```
Specially this pattern appears every time Manticore wants to compute the PC.

However, if the formula is large and there are a considerable number of solutions, sending the whole set of constraints starts to become more costly than the cost of the incremental mode